### ISSUE TITLE
rcm: "Error: Unable to get versions" using `brew livecheck --installed`

### ISSUE THREAD

**Issue by ChrisBaker97** [Homebrew/homebrew-core#64501]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

Locate potentially outdated formula using `brew livecheck --installed --newer-only` in order to submit PRs to bump them if necessary.

## What happened (include command output)

Several of my formulae return "Error: Unable to get versions" during this check. However, the only one that seems to work correctly when called by name but not via the `--installed` command line option is `rcm`. Running `brew livecheck rcm` and `brew livecheck --newer-only rcm` work as expected. However, when queried as part of the entire installed formula base via `brew livecheck --installed --newer-only`, `rcm` throws an error that it's "Unable to get versions."

<details>
  <summary>Command output</summary>
  <pre>

  $ brew livecheck --installed --newer-only
  ...
  Error: openjdk: Unable to get versions
  ...
  Error: rcm: Unable to get versions
  Error: rdfind: Unable to get versions
  ...

  $ brew livecheck openjdk rcm rdfind
  Error: openjdk: Unable to get versions
  rcm : 1.3.4 ==> 1.3.4
  Error: rdfind: Unable to get versions

  $ brew livecheck --newer-only openjdk rcm rdfind
  Error: openjdk: Unable to get versions
  Error: rdfind: Unable to get versions
  No newer upstream versions.

  </pre>
</details>

## What you expected to happen

I would've expected `rcm` not to appear at all in the `brew livecheck --installed --newer-only` output, since it's up to date.

## Step-by-step reproduction instructions (by running `brew install` commands)

```
brew install rcm
brew livecheck --installed --newer-only
brew livecheck rcm
brew livecheck --newer-only rcm
```

## Output of `brew gist-logs`, `brew config`, and `brew doctor` commands

<pre>
$ brew gist-logs rcm
Error: No logs.
$ brew config
HOMEBREW_VERSION: 2.5.9-39-g49bb256
ORIGIN: https://github.com/Homebrew/brew
HEAD: 49bb2561aad7c1956ecfb9391bb924b81df23718
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f7ad453cd337eb7351a9633643397a64952c80be
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BROWSER: open
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 11.0 build 1100
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 9.0.4, 1.8.0_202
macOS: 10.14.6-x86_64
CLT: 11.3.1.0.1.1576735732
Xcode: 11.3.1
$ brew doctor
Your system is ready to brew.
</pre>

**Comment by Rylan12** [comment]
Is it possible that you ran `brew update` in between the `--installed` and non`--installed` runs?

The `livecheck` error for `rcm` was fixed in 2707f5f21bf2875388e79fb907a946446a5e4718. Your `brew config` indicates that you've checked out a more recent commit. Before that commit, I can reproduce the error but it is present when running _both_ `brew livecheck --installed` and `brew livecheck rcm`. After that commit, I can't reproduce the error in either situation.

**Comment by ChrisBaker97** [comment]
> Is it possible that you ran `brew update` in between the `--installed` and non`--installed` runs?

Nope, but I'll tell you what is possible. I was still on the `thoughtbot` tap's `homebrew-rcm` version 1.3.3, which I discovered when I went to reinstall the formula. Thanks for taking the time to help troubleshoot. 🤦‍♂️
